lib/utils.pm

```diff
@@ -476,17 +476,17 @@ C<$params> other ar subcommand parameters, optional
 
 Examples:
     zypper_ar('http://dist.nue.suse.com/ibs/QA:/Head/SLE-15-SP1', name => 'qa-head);
-    zypper_ar('https://download.opensuse.org/repositories/devel:/kubic/openSUSE_Tumbleweed/devel:kubic.repo', no_gpg_check => priority => 90);
+    zypper_ar('https://download.opensuse.org/repositories/devel:/kubic/openSUSE_Tumbleweed/devel:kubic.repo', no_gpg_check => 1, priority => 90);
 =cut
 sub zypper_ar {
     my ($url, %args) = @_;
     my $name         = $args{name}         // '';
-    my $priority     = $args{priority}     // '';
+    my $priority     = $args{priority}     // undef;
     my $params       = $args{params}       // '';
     my $no_gpg_check = $args{no_gpg_check} // '';
 
-    $priority     = "-p $priority"  if ($priority);
-    $no_gpg_check = "--no-gpgcheck" if ($no_gpg_check);
+    $priority     = defined($priority) ? "-p $priority"  : "";
+    $no_gpg_check = $no_gpg_check      ? "--no-gpgcheck" : "";
```

Comment:
I'm blind, don't see what is wrong on previous version. Some warning I overlooked?

Reply:
try use `zypper_ar(... , no_gpg_check => 0)` :D

Reply:
if `$priority == 0`: there will be no `-p 0` as the condition will be false
if `$no_gpg_check == 0` (false): the value (in this case `0`) will be added to the zypper string

Reply:
OK, didn't know that `0` is valid value as priority (I'll read man better next time). Thx for fix.

Reply:
Even in the opposite case, using `priority => 0` will end up as `"--gpg-auto-import-keys ar -f 0 $no_gpg_check $params $url"` (same case as the gpg_check).

Reply:
man zypper: `-p 0 will set the priority back to the default (99)`